### PR TITLE
Clear Memory Buffer on Acknowledgements

### PR DESCRIPTION
**Pull request by chaitanyaphalak**

### CODE REVIEW

**Comment by chaitanyaphalak** [comment]
@luckyj5 i left them commented as they are too noisy to leave in debug for users to make sense out of.
It can be uncommented and will be useful for devs primarily.

**Comment by luckyj5** [comment]
Makes sense.

File: src/main/java/com/splunk/kafka/connect/KafkaRecordTracker.java [hunk 1]
```diff
@@ -22,21 +22,67 @@
...
+        final List<Event> events = batch.getEvents();
+        final Event event = events.get(0);
+        if (event.getTied() instanceof SinkRecord) {
+            final SinkRecord record = (SinkRecord) event.getTied();
+            TopicPartition tp = new TopicPartition(record.topic(), record.kafkaPartition());
+            //log.debug("Processing topic {} partition {}", record.topic(), record.kafkaPartition());
+            TreeMap<Long, EventBatch> tpRecords = all.get(tp);
+            if (tpRecords == null) {
+                log.error("KafkaRecordTracker removing a batch in an unknown partition {} {} {}", record.topic(), record.kafkaPartition(), record.kafkaOffset());
+                return;
```
Comment (cheneysplunk) [inline src/main/java/com/splunk/kafka/connect/KafkaRecordTracker.java:63]: should this be continue or return?

File: src/main/java/com/splunk/kafka/connect/KafkaRecordTracker.java [hunk 2]
```diff
@@ -22,21 +22,67 @@
...
+                    offset = e.getKey();
+                    iter.remove();
+                    total.decrementAndGet();
+                } else {
+                    break;
+                }
+            }
+            if (offset >= 0) {
+                if (offsets.containsKey(tp)) {
+                    offsets.replace(tp, new OffsetAndMetadata(offset + 1));
```
Comment (cheneysplunk) [inline src/main/java/com/splunk/kafka/connect/KafkaRecordTracker.java:80]: If we check the key exists, there is no differences between replace & put? I believe one `put` is enough.

Comment (chaitanyaphalak) [reply src/main/java/com/splunk/kafka/connect/KafkaRecordTracker.java:80]: oh yeah, in this case only one put will be sufficient

File: src/main/java/com/splunk/kafka/connect/KafkaRecordTracker.java (hunk 1 above)
Comment (chaitanyaphalak) [reply src/main/java/com/splunk/kafka/connect/KafkaRecordTracker.java:63]: return is fine as we cannot find the partition associated with the current record, so we skip this record.

Comment (cheneysplunk) [reply src/main/java/com/splunk/kafka/connect/KafkaRecordTracker.java:63]: Never mind. I thought it's in a loop but actually it's not.

**Pull request merged**